src/ActivitiesSDKAdapter.js

```diff
@@ -18,5 +22,66 @@ import {ActivitiesAdapter} from '@webex/component-adapter-interfaces';
 export default class ActivitiesSDKAdapter extends ActivitiesAdapter {
   constructor(datasource) {
     super(datasource);
+
+    this.activityObservables = {};
+  }
+
+  /**
+   * Returns a promise to activity data from Webex.
+   *
+   * @param {string} activityID ID of the activity for which to fetch data
+   * @returns {Promise.<Activity>} Information about the activity of the given ID
+   *
+   * @private
+   */
+  fetchActivity(activityID) {
+    logger.debug('ACTIVITY', activityID, 'fetchActivity()', ['called with', {activityID}]);
+
+    return this.datasource.messages.get(activityID);
+  }
+
+  /**
+   * Returns an observable that emits activity data of the given ID.
+   *
+   * @param {string} ID ID of activity to get
+   * @returns {external:Observable.<Activity>} Observable stream that emits activity data
+   */
+  getActivity(ID) {
+    logger.debug('ACTIVITY', ID, 'getActivity()', ['called with', {ID}]);
+
+    if (!(ID in this.activityObservables)) {
+      // use ReplaySubject cause we don't need to set an initial value
+      this.activityObservables[ID] = new ReplaySubject(1);
+
+      defer(() => this.fetchActivity(ID)).pipe(
+        map(({
+          id,
+          roomId,
+          text,
+          personId,
+          attachments,
+          created,
+        }) => ({
+          ID: id,
+          roomID: roomId,
+          text,
+          personID: personId,
+          attachments,
+          card: attachments && attachments[0] && attachments[0].contentType === 'application/vnd.microsoft.card.adaptive' && attachments[0].content,
```

Comment:
this is correct, but there are 3 possible return values for `card` using this expression:
- the card itself
- undefined (if no attachments or no first attachment)
- false (if there is an attachment which is not a card)

it would be slightly better to use a ternary operator, and return either the card or undefined.